crates/api/src/lib.rs

```diff
@@ -22,18 +24,33 @@ pub trait Perform {
 }
 
 /// Converts the captcha to a base64 encoded wav audio file
-pub(crate) fn captcha_as_wav_base64(captcha: &Captcha) -> String {
+pub(crate) fn captcha_as_wav_base64(captcha: &Captcha) -> Result<String, LemmyError> {
   let letters = captcha.as_wav();
 
-  let mut concat_letters: Vec<u8> = Vec::new();
-
+  // Decode each wav file, concatenate the samples
+  let mut concat_samples: Vec<i16> = Vec::new();
+  let mut any_header: Option<wav::Header> = None;
   for letter in letters {
-    let bytes = letter.unwrap_or_default();
-    concat_letters.extend(bytes);
+    let mut cursor = Cursor::new(letter.unwrap_or_default());
+    let (header, samples) = wav::read(&mut cursor)?;
+    any_header = Some(header);
+    let samples16 = samples.as_sixteen();
+    if samples16.is_none() {
+      return Err(LemmyError::from_message("couldnt_create_audio_captcha"));
+    }
+    concat_samples.extend(samples16.unwrap());
+  }
+
+  // Encode the concatenated result as a wav file
+  let mut output_buffer = Cursor::new(vec![]);
+  if any_header.is_none() {
+    return Err(LemmyError::from_message("couldnt_create_audio_captcha"));
   }
+  let _ = wav::write(any_header.unwrap(),
+                     &wav::BitDepth::Sixteen(concat_samples),
+                     &mut output_buffer);
```

Comment:
Are you ignoring an error here? Might have to put `?` instead.

Also you need to run `cargo +nightly fmt --all`

Reply:
Ah, good catch. Handled that error too, and formatted.